modules/sagemaker/sagemaker-templates-service-catalog/templates/model_deploy/seed_code/deploy_app/deploy_app/pipeline_stack.py

```diff
@@ -130,37 +133,41 @@ def create_inline_policy(scope: Construct, identifier: str) -> iam.Policy:
 
 
 class PipelineStack(cdk.Stack):
-    def __init__(self, scope: Construct, construct_id: str, **kwargs: Any) -> None:
-        super().__init__(scope, construct_id, **kwargs)
-
+    def create_codepipeline_with_codeccommit(self, synth_codebuild_role: iam.Role, codepipeline_role: iam.Role):
         repository = codecommit.Repository.from_repository_name(
             self, "Repository", repository_name=f"{constants.PROJECT_NAME}-deploy"
         )
 
-        codepipeline_role = iam.Role(
-            self,
-            "CodePipelineRole",
-            assumed_by=iam.ServicePrincipal("codepipeline.amazonaws.com"),
-            path="/service-role/",
-        )
-
-        codepipeline_role.attach_inline_policy(create_inline_policy(self, "DeployPipelinePolicy"))
-
-        synth_codebuild_role = iam.Role(
+        pipeline = CodePipeline(
             self,
-            "SynthStageRole",
-            assumed_by=iam.ServicePrincipal("codebuild.amazonaws.com"),
-            path="/service-role/",
+            "Pipeline",
+            pipeline_name=f"{constants.PROJECT_NAME}-pipeline",
+            synth=CodeBuildStep(
+                "Synth",
+                input=CodePipelineSource.code_commit(repository=repository, branch="main"),
+                install_commands=[
+                    "npm install -g aws-cdk",
+                ],
+                commands=["python -m pip install -r requirements.txt", 'cdk synth --app "python app.py" '],
+                role=synth_codebuild_role,
+                env=ENV,
+            ),
+            cross_account_keys=True,
+            self_mutation=True,
+            role=codepipeline_role,
         )
-        synth_codebuild_role.attach_inline_policy(create_inline_policy(self, "SynthStagePolicy"))
+        return pipeline
 
+    def create_codepipeline_with_github(self, synth_codebuild_role: iam.Role, codepipeline_role: iam.Role):
```

Comment:
Looks like the pipeline itself is the same in both cases, the only difference is the source. Does it make more sense to have `create_github_source` / `create_codecommit_source` and pass the source to the pipeline?

Reply:
Optimised